test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -349,3 +350,54 @@
                          :else (do
                                  (Thread/sleep 500)
                                  (recur (dec times)))))))))))
+
+
+
+(deftest ^:integration revoke-compile-master-test
+  (testing "Revoking compile master certificate"
+    (logutils/with-test-logging
+      (bootstrap/with-puppetserver-running-with-mock-jrubies
+       "JRuby mocking is safe here because all of the requests are to the CA
+       endpoints, which are implemented in Clojure."
+        app
+        ;; Q: Does one need to make up other config params like for eg. infra-crl-path
+        ;; or can we assume anything unspecified will exist with appropriate defaults
+        ;; Would the below override whatever defaults?
+        {:certificate-authority {:compile-masters ["dummy-compile-master"]}}
+        (testing "should update infrastructure CRL"
+          (let [ca-cert (bootstrap/get-ca-cert-for-running-server)
+                subject "dummy-compile-master"
+                ;; In one of the other unit tests I noticed the localhost cert is
+                ;; being revoked. If that does not affect things we could do that instead of 
+                ;; creating a new one. But wondering it actually may have impact on other 
+                ;; tests since this is integration
+                cm-cert (bootstrap/get-cert-signed-by-ca-for-running-server
+                             ca-cert
+                             subject)
+                 ;; TODO - Extract serial number of abvoe cert. Is there an easy way
+                 ;; to do it here or can we make assumption on the expected serial number
+                 ;;
+
+                 ;; Q: Do we need to do something to explicitly moved the certificate
+                 ;; to CA's signed folder
```

Comment:
The CA writes a copy of the cert out to the signed dir when it signs it.

Reply:
Yes I know (when a cert is issued over proper interfaces) but was not sure the method being used in test would do it.

Reply:
Hmm.. looks like this particular method - bootstrap/get-cert-signed-by-ca-for-running-server - used to sign the cert does not place the cert in signed folder. 
Got this travis error: 
  actual: java.io.FileNotFoundException: ./target/master-conf/ssl/ca/signed/dummy-compile-master.pem (No such file or directory)